airbyte-integrations/connectors/source-postgres/src/test-integration/java/io/airbyte/integrations/io/airbyte/integration_tests/sources/CdcPostgresSourceComprehensiveTest.java

```diff
@@ -0,0 +1,539 @@
+            .addExpectedValues("101", null)
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("boolean")
+            .airbyteType(JsonSchemaPrimitive.BOOLEAN)
+            .addInsertValues("true", "'yes'", "'1'", "false", "'no'", "'0'", "null")
+            .addExpectedValues("true", "true", "true", "false", "false", "false", null)
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("bytea")
+            .airbyteType(JsonSchemaPrimitive.OBJECT)
+            .addInsertValues("decode('1234', 'hex')")
+            .addExpectedValues("EjQ=")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("character")
+            .airbyteType(JsonSchemaPrimitive.STRING)
+            .addInsertValues("'a'", "'*'", "null")
+            .addExpectedValues("a", "*", null)
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("character")
+            .fullSourceDataType("character(8)")
+            .airbyteType(JsonSchemaPrimitive.STRING)
+            .addInsertValues("'{asb123}'", "'{asb12}'")
+            .addExpectedValues("{asb123}", "{asb12} ")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("varchar")
+            .airbyteType(JsonSchemaPrimitive.STRING)
+            .addInsertValues("'a'", "'abc'", "'Миші йдуть на південь, не питай чому;'", "'櫻花分店'",
+                "''", "null", "'\\xF0\\x9F\\x9A\\x80'")
+            .addExpectedValues("a", "abc", "Миші йдуть на південь, не питай чому;", "櫻花分店", "",
+                null, "\\xF0\\x9F\\x9A\\x80")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("varchar")
+            .fullSourceDataType("character(12)")
+            .airbyteType(JsonSchemaPrimitive.STRING)
+            .addInsertValues("'a'", "'abc'", "'Миші йдуть;'", "'櫻花分店'",
+                "''", "null")
+            .addExpectedValues("a           ", "abc         ", "Миші йдуть; ", "櫻花分店        ",
+                "            ", null)
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("cidr")
+            .airbyteType(JsonSchemaPrimitive.STRING)
+            .addInsertValues("null", "'192.168.100.128/25'", "'192.168/24'", "'192.168.1'",
+                "'128.1'", "'2001:4f8:3:ba::/64'")
+            .addExpectedValues(null, "192.168.100.128/25", "192.168.0.0/24", "192.168.1.0/24",
+                "128.1.0.0/16", "2001:4f8:3:ba::/64")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("date")
+            .airbyteType(JsonSchemaPrimitive.STRING)
+            .addInsertValues("null")
+            .addNullExpectedValue()
+            .build());
```

Comment:
Nit: this case is included in the one below. So it seems redundant.

Reply:
removed redundant, thanks